### PR TITLE
Feature/scatterplot

### PR DESCRIPTION
**Pull request by kzadurska**
Zrzut ekranu 2022-02-28 o 16 03 53
Zrzut ekranu 2022-02-28 o 16 07 42
Zrzut ekranu 2022-02-28 o 16 04 31

### CODE REVIEW

File: src/client/visualizations/scatterplot/scatterplot.tsx [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+      yTicks,
+      xScale,
+      yScale,
+      xSeries,
+      ySeries,
+      plottingStage,
+      scatterplotData
+    } = this.getPlottingData(data, essence, stage);
+
+    const rightXAxisLabelPosition = stage.width - (plottingStage.width + plottingStage.x);
```
Comment (kzadurska) [inline src/client/visualizations/scatterplot/scatterplot.tsx:80]: > Candidate for extracted method with unit test.

https://github.com/allegro/turnilo/pull/841#discussion_r811649776

File: src/client/visualizations/scatterplot/tooltip.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  splitKey,
+  timezone,
+  showPrevious
+}) => {
+  if (!isTruthy(datum)) return null;
+
+  const xValue = xSeries.selectValue(datum);
+  const yValue = ySeries.selectValue(datum);
+
+  return <TooltipWithinStage top={yScale(yValue) + TOOLTIP_OFFSET_Y} left={xScale(xValue) + TOOLTIP_OFFSET_X} stage={stage}>
```
Comment (kzadurska) [inline src/client/visualizations/scatterplot/tooltip.tsx:62]: Extract calculation to line above

> Extract calculation, xValue and yValue could be defined within offsets in lines 44/45.
https://github.com/allegro/turnilo/pull/841#discussion_r811652300

File: src/client/visualizations/scatterplot/scatterplot.scss [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+      &.axis-label-x {
+        text-anchor: middle;
+      }
+    }
+  }
+
+  .point {
+    @include css-variable(stroke, brand);
+    @include css-variable(fill, item-dimension);
```
Comment (adrianmroz-allegro) [inline src/client/visualizations/scatterplot/scatterplot.scss:52]: That's why we need viz colours in AppSettings. #852

File: src/common/models/series-list/series-list.ts [hunk 4]
```diff
@@ -140,10 +140,18 @@ export class SeriesList extends Record<SeriesListValue>(defaultSeriesList) {
     return this.updateSeries(series => series.take(1));
   }
 
+  public takeNFirst(number: number) {
```
Comment (adrianmroz-allegro) [inline src/common/models/series-list/series-list.ts:143]: Missing return type

File: src/common/visualization-manifests/scatterplot/scatterplot.ts [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+import { Split } from "../../models/split/split";
+import { Resolve, VisualizationManifest } from "../../models/visualization-manifest/visualization-manifest";
+import { Predicates } from "../../utils/rules/predicates";
+import {
+  ActionVariables,
+  visualizationDependentEvaluatorBuilder
+} from "../../utils/rules/visualization-dependent-evaluator";
+import { settings } from "./settings";
+
+const rulesEvaluator = visualizationDependentEvaluatorBuilder
```
Comment (adrianmroz-allegro) [inline src/common/visualization-manifests/scatterplot/scatterplot.ts:28]: ❤️

**Review by adrianmroz-allegro — APPROVED** [review]

**Review by mkuthan — APPROVED** [review]
Version deployed from this branch works like a charm, thanks!

File: src/client/visualizations/scatterplot/tooltip.tsx (hunk 2 above)
Comment (kzadurska) [reply src/client/visualizations/scatterplot/tooltip.tsx]: Moved to lines above d99a7db8b8479b0829f1c5a877f3bc701c3a2eab

File: src/client/visualizations/scatterplot/scatterplot.tsx (hunk 1 above)
Comment (kzadurska) [reply src/client/visualizations/scatterplot/scatterplot.tsx]: Extracted d99a7db8b8479b0829f1c5a877f3bc701c3a2eab

File: src/client/visualizations/scatterplot/utils/get-plotting-data.mocha.ts [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    const extent = [10, 100];
+
+    const result = extendExtentIfNeeded(extent);
+
+    expect(result).to.eql([10, 100]);
+  });
+});
+
+describe("getTicksForAvailableSpace", () => {
+  it("should return not changed ticks when screen size is larger than 768px", () => {
```
Comment (adrianmroz-allegro) [inline src/client/visualizations/scatterplot/utils/get-plotting-data.mocha.ts:39]: Not "screen size" but general "available space". "when has more than 768px space available"?

Comment (kzadurska) [reply src/client/visualizations/scatterplot/utils/get-plotting-data.mocha.ts:39]: Yes, that's more accurate. Will fix!

Comment (kzadurska) [reply src/client/visualizations/scatterplot/utils/get-plotting-data.mocha.ts]: aefaeec ✅

**Pull request merged**